test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/SortTests.cs

```diff
@@ -480,6 +480,136 @@ public async Task GivenQueryWithObservation_WhenSearchedForItemsWithNoSubjectAnd
             await ExecuteAndValidateBundle($"Observation?_tag={tag}&_sort=_lastUpdated&subject:missing=true", false, expected_resources.ToArray());
         }
 
+        [SkippableFact]
+        public async Task GivenPatientsWithMultipleNames_WhenFilteringAndSortingByFamilyName_ThenResourcesAreReturnedInAscendingOrder()
+        {
+            var tag = Guid.NewGuid().ToString();
+            var patients = await CreatePatientsWithMultipleFamilyNames(tag);
+
+            await ExecuteAndValidateBundle($"Patient?_tag={tag}&family=R&_sort=family", sort: false, patients[0..5]);
+        }
+
+        [SkippableTheory]
+        [InlineData(2)]
+        [InlineData(3)]
+        [InlineData(4)]
+        public async Task GivenPatientsWithMultipleNamesAndPaginated_WhenFilteringAndSortingByFamilyName_ThenResourcesAreReturnedInAscendingOrder(int count)
+        {
+            var tag = Guid.NewGuid().ToString();
+            var patients = await CreatePatientsWithMultipleFamilyNames(tag);
+
+            await ExecuteAndValidateBundle($"Patient?_tag={tag}&family=R&_sort=family&_count={count}", sort: false, pageSize: count, patients[0..5]);
+        }
+
+        [Fact]
+        [HttpIntegrationFixtureArgumentSets(dataStores: DataStore.SqlServer)]
```

Comment:
Family should be supported in Cosmos by default, whats the diff for this?

Reply:
The diff here is in the way we break ties. For Cosmos, we always choose the last updated resource (irrespective of the overall sort order). For SQL, we choose the resource based on overall sort order. Hence we see a difference when sorting by Descending order. Have added another test for Cosmos.

Reply:
Just for my understanding, why is this different :D? Isn't the secondary sort ResourceSurrogateId, which is in a sense LastUpdated?

Reply:
Sorry, I messed up the explanation. I meant to say "oldest" last updated resource (instead of last updated resource). And I had the behavior reversed.

Cosmos always chooses the resource based on last updated time ordered by overall sort order. For SQL, we always choose the "oldest" resource based on last updated time i.e. ResourceSurrogateId ASC